pandas/io/sas/sasreader.py

```diff
@@ -14,16 +14,61 @@
 )
 
 from pandas._typing import (
+    CompressionOptions,
     FilePath,
     ReadBuffer,
 )
+from pandas.util._decorators import (
+    Appender,
+    deprecate_nonkeyword_arguments,
+)
+
+from pandas.core.shared_docs import _shared_docs
 
 from pandas.io.common import stringify_path
 
 if TYPE_CHECKING:
     from pandas import DataFrame
 
 
+_doc_read_sas = r"""
```

Comment:
Seems strange to pull this out just to use `Appender`.  Why not leave it in-line and use `Substitution`?

Reply:
See, e.g., https://github.com/pandas-dev/pandas/blob/c5e32b560444b15fb6e4bf03366df3df415f8adf/pandas/core/groupby/groupby.py#L2425

Reply:
Thanks, I copy pasted that from some other place.

Reply:
@bashtage now we have the problem that the substituted snippet is dedented:

```
...
iterator : bool, defaults to False
    If True, returns an iterator for reading the file incrementally.

    .. versionchanged:: 1.2

        ``TextFileReader`` is a context manager.
compression : str or dict, default 'infer'
For on-the-fly decompression of on-disk data. If 'infer' and '%s' is
...
```

Last line should be indented.

Any pointers how to deal with this?

Reply:
Maybe use the `@doc` decorator as in here

https://github.com/pandas-dev/pandas/blob/fa7e31b8b19eb03dceebd09d03798363daae07d9/pandas/io/xml.py#L49